Sample for fetchContent (#1392)
Azure/azure-sdk-for-cpp: samples/Integration/CMake-fetch-content-app/README.md

> # Integrating the Azure SDK for C++ with CMake fetch content to your application

Reviewer: ```suggestion
# Integrating the Azure SDK for C++ into your application using CMake fetch content
```

> This application is an example of integrating the Azure SDK for C++ to your application. It uses the CMake fetch content functionality to automatically clone the Azure SDK for C++ repo to your `build` directory. This approach is useful if you are using CMake to build your application.

Reviewer 2: ```suggestion
This application shows how to integrate the Azure SDK for C++ in your application. It uses CMake fetch content functionality to clone the Azure SDK for C++ repo into your `build` directory. This approach is useful when using CMake to build your application.
```

## Pre-requisites

> Make sure to install the [Azure SDK for C++ dependencies](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#third-party-dependencies) first.

Reviewer 2: ```suggestion
Install the [Azure SDK for C++ dependencies](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#third-party-dependencies).
```

> - CMake project.

Reviewer 2: Specify the version of CMake that is required, add links to CMake for those that don't have it installed

- C++ version 14 or greater.

## Build

```bash
#
# Building the application.
# Instructions from application root directory.
#

# Create build directory just the first time.
mkdir build
cd build

# Generate and build
# This code assumes that the SDK dependencies were installed with VCPKG

> cmake -DCMAKE_TOOLCHAIN_FILE=path/to/vcpkg/scripts/buildsystems/vcpkg.cmake ..

Reviewer: Is this the recommended (and only) way to let cmake know to pickup vcpkg or is there a way to set it up, so I don't have to repeatedly pass in `CMAKE_TOOLCHAIN_FILE`?

Author: it's a personal/developer preference.

Some projects do what the Azure SDK does and set the toolchain inside the CMake list. The bad thing about this is if someone wants to set a different value, they can't use command line, they need to update the CMake file.
The benefit is to avoid writing this command, however, you tipycally only need to write it once, then you would use the commands history to repeat.

If we see what VS does, it generates and set the toolchain as part of the generate script every time.

So, I don't think there's a right or wrong way to do this.  

For this case, I just want to make sure that the sample can be copy pasted and it would build/run

cmake -build .
```

## Run application

> Review source code header for `environment variables` that must be set up before running the app.

Reviewer: What happens if I run it with none of the environment variables set? Does it tell me which ones I need to set?

Author: naaa,  it crashes because when we call getenv() and it doesn't fin the variable, there's an exception.

> Review source code header to learn if there is any `environment variable` that needs to be set up before running the app.

Reviewer 2: ```suggestion
Review source code header for `environment variables` that must be set up before running the app.
```

```bash
#
# Running the Application
# Instructions from inside the build directory.
#

# Run binary (.exe on Windows)

> ./applicationName

Reviewer: Is this the name of the exe that gets built? Where is this defined? Is it not `application`?

Author: Tt is `application`, yes, 
it is defined in the CMake file with the `add_executable` instruction.

I used this name to avoid a hard relationship between the Readme and the code.  Then I the name in the sample can change in the future and this is still valid

Reviewer: I don't think having a hard relationship is much of a concern here since we shouldn't really be changing the samples all that much.

The way it is written now is more confusing to follow along (especially since there is no indication where this is defined or what the name ought to be).

Making it straight-forward to follow along and get the application running, without any changes, is valuable for adoption.
```suggestion
./application
```
